### PR TITLE
Support signed integers and `char` in v0 mangling

### PR DESCRIPTION
**Pull request by varkor**
Likely we want more tests, to check the output is correct too: however, I wasn't sure what kind of test we needed, so I just added one similar to that added in https://github.com/rust-lang/rust/pull/77452 for now.

r? @eddyb

### CODE REVIEW

File: compiler/rustc_symbol_mangling/src/v0.rs [hunk 1]
```diff
@@ -502,17 +502,27 @@ impl Printer<'tcx> for SymbolMangler<'tcx> {
...
-            ty::Uint(_) => {}
-            ty::Bool => {}
+        let mut neg = false;
+        let val = match ct.ty.kind() {
+            ty::Uint(_) | ty::Bool | ty::Char => {
+                ct.try_eval_bits(self.tcx, ty::ParamEnv::reveal_all(), ct.ty)
+            }
+            ty::Int(_) => {
+                neg = true;
+                ct.val
```
Comment (varkor) [inline compiler/rustc_symbol_mangling/src/v0.rs:512]: I'm not sure whether this is the right approach.

Comment (eddyb) [reply compiler/rustc_symbol_mangling/src/v0.rs:512]: `neg = true` doesn't seem right when the value could be positive.

Comment (varkor) [reply compiler/rustc_symbol_mangling/src/v0.rs:512]: I should stop writing code after midnight…

**Comment by varkor** [comment]
Blocked on https://github.com/alexcrichton/rustc-demangle/pull/39.

**Comment by varkor** [comment]
@eddyb: this is now ready to review.

**Review by eddyb — APPROVED** [review]
r=me

**Comment by varkor** [comment]
I'll rebase on top of https://github.com/alexcrichton/rustc-demangle/pull/41 as soon as it's merged.

**Pull request merged**